### ISSUE TITLE
$preserveKeys does not work for ResourceClass::collection()

### ISSUE THREAD

**Issue by Organizm238** [laravel/framework#28653]
- Laravel Version: 5.8.4
- PHP Version: 7.1
- Database Driver & Version: mysql

### Description:

According to the documentation (https://laravel.com/docs/5.8/eloquent-resources#resource-collections , chapter 'Preserving Collection Keys'), to preserve keys in resource collection, you just need to set

```
public $preserveKeys = true;
```
in resource class and it should work for `ResourceClass::collection($items)`. But it doesn't. It works only for custom resource collections. 

After some debugging: seems like we are running this check (`ConditionallyLoadsAttributes.php@removeMissingValues`):
```
       if (property_exists($this, 'preserveKeys') && $this->preserveKeys === true) {
            return $data;
        }
```
on `AnonymousResourceCollection.php` class, not on a Resource class. That's why `preserveKeys` property doesn't exists. 

However, if we use custom resource collection with overriding default `toArray()` method (`ResourceCollection@toArray`):
```
public function toArray($request)
    {
        return $this->collection->map->toArray($request)->all();
    }
```
with something like:
```
public function toArray($request)
    {
        return [
            'data' => $this->collection,
        ];
    }
```

, the `property_exists($this, 'preserveKeys')` check is run on Resource class and everything works fine.

### Steps To Reproduce:

Just reproduce the example from the doc:
1. Have (or add) some users in DB

2. Add `UserResource` : 
```
class UserResource extends JsonResource
{
    public $preserveKeys = true;

    public function toArray($request)
    {
        return [
            'id' => $this->id,
        ];
    }
}
```

3. Add route to `web.php`:
```
Route::get('/test', function () {
    return UserResource::collection(User::all()->take(2)->keyBy->id);
});
```

4. Open `/test` in browser and see, that keys are not preserved. 

**Actual output**:
```
{"data":[{"id":1},{"id":2}]}
```

**Expected output**:
```
{"data":{"1":{"id":1},"2":{"id":2}}}
```

**Comment by staudenmeir** [comment]
This has been fixed in Laravel 5.8.9.

**Comment by driesvints** [comment]
Please upgrade to the latest version, thanks.
